src/coreclr/gc/gc.cpp

```diff
@@ -11856,7 +11856,13 @@ void gc_heap::return_free_region (heap_segment* region)
     }
     clear_region_info (region);
 
+#ifndef MULTIPLE_HEAPS
+    free_list_snapshot::record(return_free_region_start, free_regions_basic, &free_regions[(int)basic_free_region]);
```

Comment:
this should only record when `oh` is `gc_oh_num::soh` because SOH and UOH allocations take different locks.